Library/Homebrew/os/mac.rb

```diff
@@ -98,31 +98,18 @@ def self.active_developer_dir
       )
     end
 
-    sig { returns(T::Boolean) }
-    def self.sdk_root_needed?
-      if MacOS::CLT.installed?
-        # If there's no CLT SDK, return false
-        return false unless MacOS::CLT.provides_sdk?
-        # If the CLT is installed and headers are provided by the system, return false
-        return false unless MacOS::CLT.separate_header_package?
-      end
-
-      true
-    end
```

Comment:
This is de-facto public API and should have a deprecation path: https://github.com/search?q=%22%3C+Formula%22+sdk_root_needed%3F+NOT+is%3Afork&type=code